rds-replica/main.tf

```diff
@@ -18,6 +18,8 @@ resource "aws_db_instance" "rds" {
   replicate_source_db             = var.replicate_source_db
   db_subnet_group_name            = aws_db_subnet_group.rds[0].id
   storage_encrypted               = var.storage_encrypted
+  allocated_storage               = var.allocated_storage ? data.aws_db_instance.master.rds_allocated_storage : var.allocated_storage
```

Comment:
setting the two attributes to the corresponding variables directly will already do the trick, as they default to `null` and thus the default resource behaviour. There's no need to refer to the master data provider.

Reply:
Removed the usage and the ouput again